src/cl/ucc_base_iface.c

```diff
@@ -0,0 +1,36 @@
+#include "ucc_base_iface.h"
+#include "utils/ucc_malloc.h"
+#include "utils/ucc_log.h"
+
+ucc_config_field_t ucc_base_config_table[] = {
+    {"LOG_LEVEL", "warn",
+     "UCC logging level. Messages with a level higher or equal to the "
+     "selected will be printed.\n"
+     "Possible values are: fatal, error, warn, info, debug, trace, data, func, "
+     "poll.",
+     ucc_offsetof(ucc_base_config_t, log_component), UCC_CONFIG_TYPE_LOG_COMP},
+
+    {NULL}};
+
+ucc_status_t ucc_base_config_read(const char *full_prefix,
```

Comment:
This looks awfully familiar... does this mean every "framework", e.g. CL/TL/..., will have this kind of function? could we merge all these functions into one?